### ISSUE TITLE
It stopped working after updating to macOS 14.4 🥹

### ISSUE THREAD

**Issue by roelvangils** [briankendall/forceFullDesktopBar#20]
[Whoops](https://app.warp.dev/block/BWFqpjwDUdJpMDCtHsw3Hu) 🙂

**Comment by briankendall** [comment]
Yup, looks like 14.4 broke it. The issue appears to be with frida. Hopefully updating it will fix the issue.

**Comment by briankendall** [comment]
Unfortunately updating Frida doesn't fix it. Apple has done something to make it more difficult to inject code into the Dock. I'll likely need to find a new method of doing the injection, other than using Frida, or wait for Frida to make a fix themselves. It could be a while until forceFullDesktopBar is going to work in 14.4 and later.

**Comment by roelvangils** [comment]
Oh, good to know. Thanks for the preliminary research! Now that it no longer works, I realise again how annoying the default way spaces are shown is! 😁 Meanwhile, I think I'll use the less elegant "fix" in BetterTouchTool (it moves the mouse cursor briefly to the top before executing Mission Control). Maybe other users of forceFullDesktopBar might like this too as a temporary fix.

**Comment by briankendall** [comment]
@roelvangils I also recommend checking out my other tool [missionControlFullDesktopBar](https://github.com/briankendall/missionControlFullDesktopBar) which I made several years ago as an alternative method that doesn't require code injection or disabling SIP. It still works in Sonoma! And its method doesn't involve moving the mouse cursor.

**Comment by roelvangils** [comment]
Thanks, @briankendall! You clearly put a lot of sweat and tears in this! 😁 The other tools also works great, but (honestly) my earlier hack works just as well (for now):

![20240311_203424_BetterTouchTool@2x](https://github.com/briankendall/forceFullDesktopBar/assets/103744/5c31b857-f0eb-49cc-975a-4b953dc687ec)

Are you planning to compile missionControlFullDesktopBar for Apple Silicon? I see it's still for Intel. It probably doesn't make any difference 😁 

![20240311_204030_Finder@2x](https://github.com/briankendall/forceFullDesktopBar/assets/103744/f23caf89-d2a6-4eb9-9fb1-07122a663bb2)

**Comment by briankendall** [comment]
I should recompile it, since it wouldn't take much effort. I don't think the app needs to be updated at all, just recompiled.

**Comment by amilios** [comment]
This is a bit of a dumb "solution", but actually you can just add a bunch of empty desktops to force Mission Control to start with the previews open. on my Air it takes only 9 virtual desktops. Unclear if this has any other sort of downside (performance penalty? I assume minimal if most of the desktops are empty :P )

**Comment by roelvangils** [comment]
> I should recompile it, since it wouldn't take much effort. I don't think the app needs to be updated at all, just recompiled.

Hi Brian, I hope you're doing fine. Have you considered recompiling the app for MacOS Sequoia? 😉

**Comment by amilios** [comment]
Works perfectly! Thank you :)
